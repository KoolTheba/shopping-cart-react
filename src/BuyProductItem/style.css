.product-image-table{
    height: 3rem;
    border-radius: 50%;
    -moz-border-radius: 50%;
    -webkit-border-radius: 50%;
    width: 3rem;
}