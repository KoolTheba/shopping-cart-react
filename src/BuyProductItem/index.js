import React from 'react'
import PropTypes from 'prop-types'
import './style.css'

import { makeStyles } from '@material-ui/core/styles'
import TableCell from '@material-ui/core/TableCell'
import TableRow from '@material-ui/core/TableRow'
import IconButton from '@material-ui/core/IconButton'
import DeleteIcon from '@material-ui/icons/Delete'
import RemoveShoppingCartIcon from '@material-ui/icons/RemoveShoppingCart'
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart'

const useStyles = makeStyles((theme) => ({
    tableBody: {
        color: '#292C34',
        fontSize: 20,
        fontFamily: 'sans-serif',
        margin: theme.spacing(1),
    }
}));

function BuyProductItem (props) {
    const classes = useStyles();

    return (
        <TableRow key={props.product.id} className='products-table'>
              <TableCell  align='center' component="th" scope="row">
                <img
                    className='product-image-table'
                    src={props.product.image}
                    alt={props.product.name}
                />
              </TableCell>
              <TableCell align='left' className={classes.tableBody}>{props.product.name}</TableCell>
              <TableCell align="center">
                <IconButton
                    className='delete-button'
                    aria-label="delete" 
                    onClick={() => props.removeListItem(props.product.id)}
                >
                    <DeleteIcon />
                </IconButton>
              </TableCell>
              <TableCell align="center" className={classes.tableBody}>{props.product.price}€</TableCell>
              <TableCell align="center" className={classes.tableBody}>
                <IconButton
                    className='remove-product-button'
                    aria-label="remove from shopping cart"
                    onClick={() => props.removeUnit(props.product.id)}
                    disabled={props.product.quantity > 1 ? false : true}    
                >
                    <RemoveShoppingCartIcon />
                </IconButton>
                {props.product.quantity}
                <IconButton 
                    className='add-product-button'
                    aria-label="add to shopping cart"
                    onClick={() => props.addUnit(props.product.id)}
                >
                    <AddShoppingCartIcon />
                </IconButton>
              </TableCell>
              <TableCell 
                    align="center"
                    className={classes.tableBody}
                    >{props.product.price*props.product.quantity}€
              </TableCell>
        </TableRow>
    )
}

BuyProductItem.propTypes = {
    product: PropTypes.object,
    addUnit: PropTypes.func,
    removeUnit: PropTypes.func,
    removeListItem: PropTypes.func
}

export default BuyProductItem