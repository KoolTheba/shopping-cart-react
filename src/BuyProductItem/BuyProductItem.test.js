import React from 'react'
import BuyProductItem from './'
import { shallow } from 'enzyme'

const props = {
    product: {
        id: "PLANT14TGG",
        quantity: 1,
        image: "https://image.shutterstock.com/image-photo/pot-zamioculcas-home-plant-on-600w-1399949501.jpg",
        name: "Zamioculcas",
        price: 90.00
    },
    removeListItem: jest.fn(),
    removeUnit: jest.fn(),
    addUnit: jest.fn()
}

describe('BuyProductItem component', () => {
    it('renders BuyProductItem component correctly', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        expect(wrapper.find('.products-table')).toHaveLength(1)
        expect(wrapper).toMatchSnapshot()
    });

    it('renders delete, remove and add button correctly', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        expect(wrapper.find('.delete-button')).toHaveLength(1)
        expect(wrapper.find('.remove-product-button')).toHaveLength(1)
        expect(wrapper.find('.add-product-button')).toHaveLength(1)
    });

    it('remove button is disable when product quantity is 1', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        const removeButton = wrapper.find('.remove-product-button').prop('disabled')
        expect(removeButton).toBe(true)
    });

    it('remove button is not disable when product quantity is more than 1', () => {
        props.product.quantity = 2
        const wrapper = shallow(<BuyProductItem {...props}/>)
        const removeButton = wrapper.find('.remove-product-button').prop('disabled')
        expect(removeButton).toBe(false)
    });

    it('when clicking delete button, removeListItem() is called', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        const deleteButton = wrapper.find('.delete-button')
        deleteButton.simulate('click')
        expect(props.removeListItem).toHaveBeenCalledTimes(1)
        expect(props.removeListItem).toHaveBeenCalledWith(props.product.id)
    });

    it('when clicking remove button, removeUnit() is called', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        const removeButton = wrapper.find('.remove-product-button')
        removeButton.simulate('click')
        expect(props.removeListItem).toHaveBeenCalledTimes(1)
        expect(props.removeUnit).toHaveBeenCalledWith(props.product.id)
    });

    it('when clicking add button, addUnit() is called', () => {
        const wrapper = shallow(<BuyProductItem {...props}/>)
        const addButton = wrapper.find('.add-product-button')
        addButton.simulate('click')
        expect(props.removeListItem).toHaveBeenCalledTimes(1)
        expect(props.addUnit).toHaveBeenCalledWith(props.product.id)
    });
});
