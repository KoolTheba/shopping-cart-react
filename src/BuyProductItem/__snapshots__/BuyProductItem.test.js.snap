// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyProductItem component renders BuyProductItem component correctly 1`] = `ShallowWrapper {}`;
